### ISSUE TITLE
PSI showing in KpA

### ISSUE THREAD

**Issue by magnetic157** [itchannel/fordpass-ha#230]
Ever since the newest update I can only get KpA and BAR to show appropriate values... PSI looks to be showing in KpA.

When I change the option to BAR it changes but not when I change it to PSI.

<img width="346" alt="image" src="https://user-images.githubusercontent.com/78054864/209997995-48e7f432-63b9-44f6-bddf-6db11b80b7b5.png">

<img width="486" alt="image" src="https://user-images.githubusercontent.com/78054864/209998058-9718216c-731a-4c00-b056-288563d56416.png">

<img width="426" alt="image" src="https://user-images.githubusercontent.com/78054864/210001388-8e11662f-4886-437f-b239-8e55c1c9ec9e.png">

**Comment by itchannel** [comment]
Did you restart HA after changing the option?

**Comment by magnetic157** [comment]
Yes, each time.  Like I said the values change between BAR and KpA but not PSI

**Comment by Game2Freak** [comment]
I’m having the exact same issue.

**Comment by stumiles86** [comment]
Having this too

**Comment by itchannel** [comment]
I can't replicate the issue on my end. Can one of you please add debugging to the fordpass integration and send me the contents of the vehicle array that is shown in the log. 

The only thing I can think of is the API is now returning the tire pressure in the chosen uniit in the app or specific to a region. e.g. Australia in Kpa. 

```
logger:
  default: warning
  logs:
    custom_components.fordpass: debug
```

**Comment by costr** [comment]
I am also having the same issue. You can see a dump of debug data here: https://gist.github.com/costr/41891cfb841a0c4cbb2a6095286a7240

**Comment by costr** [comment]
Also, if it's helpful, here's what my core.config_entries shows, for FordPass (with redactions):
```
{
  "entry_id": "1569262218e6b21fbf324692952d21a0",
  "version": 1,
  "domain": "fordpass",
  "title": "redacted title",
  "data": {
    "username": "redacted@myemail.com",
    "password": "redacted",
    "vin": "redacted",
    "region": "North America & Canada"
  },
  "options": {
    "pressure_unit": "PSI",
    "distance_unit": "mi",
    "update_interval": 900
  },
  "pref_disable_new_entities": false,
  "pref_disable_polling": false,
  "source": "user",
  "unique_id": null,
  "disabled_by": null
},
```

**Comment by itchannel** [comment]
@costr Do you mind running the new pre-release I just added. It only adds extra logging for the tire pressure function. But would be good if you could post the updated log so I can see exactly what the tire pressure function is returning. 

https://github.com/itchannel/fordpass-ha/releases/tag/1.42-Test

Thanks for the vehicle debug you posted, looks like the API returns the same values as mine so it must be something funky in the code hence my debug update.

**Comment by costr** [comment]
It looks like the elif changes have it working, in the test release. I am now seeing PSI in the values. Here's the output, either way, I've updated the original gist: https://gist.github.com/costr/41891cfb841a0c4cbb2a6095286a7240

**Comment by costr** [comment]
Oh, yeah, that makes sense, if looks like there were two if statements, but the second one had an else, for falling back to KpA, but since the second if was for "BAR" it would fall back to KpA. Good on ya! Thanks for working with us on this one and for a great extension!

**Comment by magnetic157** [comment]
@costr @itchannel thank you both for the quick resolution!!

**Comment by itchannel** [comment]
Thanks @costr for checking. I did wonder if the If statement was the issue which is why I changed it but weird that it never caused me an issue. 👍 

I'll push the 1.42 release later today for everyone else.

**Comment by itchannel** [comment]
Fixed in 1.42 Release.
